### PR TITLE
print versionless messages after resolution

### PR DESCRIPTION
**Pull request by rsherget**
Use messages after feature resolution to display versionless specific information/errors.

- [ ] Considered risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes)
- [ ] Resolved Issues: Fixes #FILLMEIN... (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions)
- [ ] Resolved external Known Issues (including APARS): Fixes #FILLMEIN...

### CODE REVIEW

**Comment by rsherget** [comment]
Build: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_p9mccECBEe-Hn95lLM_tGQ

**Comment by rsherget** [comment]
Build: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_mP9B8EFxEe-F88sHKWQ5jw

**Review by cbridgha — APPROVED** [review]
Thanks - looks much better than the draft!

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureResolverImpl.java
Comment (tbitonti) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureResolverImpl.java:1]: Around line 274:
if(duplicates.containsKey(parseName(platformFeature.getSymbolicName()))){
There are lots of calls to parseName which can be stored in a local.
Does parseName always return a lower case name?

Comment (tbitonti) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureResolverImpl.java:1]: Having a local reference to 'context.getResult()' would help streamline 'finalizeVersionlessResults'.
That there are so many references to the result indicates that the behavior might be better placed in the result object.  (That may or may not be possible, depending on what types we access in Result.)

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/resolver/FeatureResolver.java
Comment (tbitonti) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/resolver/FeatureResolver.java:1]: These need comments.

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureResolverResultImpl.java
Comment (tbitonti) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureResolverResultImpl.java:1]: Some notes on what are the keys and values of the new results collections, and what is the meaning of the new collections should be provided.

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java
Comment (tbitonti) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java:1]: Lines 984-998 could be a method on the result object.
Need a comment to explain that the lookups are to change the resolved name to the feature name.
Do we want to sort the 'resolvedVersionless' and 'resolvedVersioned'?  (Hmm, maybe not -- that might change the order of one of the lists and the lists wouldn't line up anymore.

Comment (tbitonti) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java:1]: Near line 1759, in 'areConfiguredFeaturesGood', the second reference to 'isBeta' is redundant:
            if( !isBeta || (**isBeta** && featureRepository.getPlatforms().equals(newConfiguredPlatforms)

**Review by tbitonti — COMMENTED** [review]
Don't see anything broken, just some cleanups.

**Pull request merged**